core/server/api_container/server/api_container_service.go

```diff
@@ -651,3 +686,64 @@ func (apicService ApiContainerService) runStarlark(
 		}
 	}
 }
+
+func getFileDescriptionsFromArtifact(artifactPath string) ([]*kurtosis_core_rpc_api_bindings.FileArtifactContentsFileDescription, error) {
+	file, err := os.Open(artifactPath)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "Failed to get file descriptions for artifact path '%v'", artifactPath)
+	}
+	defer file.Close()
+
+	gzipReader, err := gzip.NewReader(file)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "Failed to get create gzip reader for artifact path '%v'", artifactPath)
+	}
+	defer gzipReader.Close()
+
+	tarReader := tar.NewReader(gzipReader)
+
+	fileDescriptions := []*kurtosis_core_rpc_api_bindings.FileArtifactContentsFileDescription{}
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, stacktrace.Propagate(err, "Failed to get header from artifact path '%v'", artifactPath)
+		}
+
+		filePath := header.Name
+		description := fmt.Sprintf("Size: %v", header.Size)
+		textPreview, err := getTextRepresentation(tarReader, unlimitedLineCount)
+		if err != nil {
+			logrus.Debugf("Failed to get text preview for file '%v' with error '%v'", filePath, textPreview)
```

Comment:
i think it would be good to return the file-artifacts that errored out as well; though can be done as a FLUP.

Reply:
Added a TODO